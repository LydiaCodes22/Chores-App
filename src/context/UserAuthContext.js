import React, { createContext, useContext, useEffect, useState } from "react";
import {
  createUserWithEmailAndPassword,
  signInWithEmailAndPassword,
  onAuthStateChanged,
  signOut,
} from "firebase/auth";
import { auth } from "../firebase-config";

export const userAuthContext = createContext();

// eslint-disable-next-line react/prop-types
export const UserAuthContextProvider = ({ children }) => {
  const [user, setUser] = useState({});
  const [chores, setChores] = useState([]);
  const [familyID, setFamilyID] = useState(null);
  const [userID, setUserID] = useState(null);
  const [userRole, setUserRole] = useState(null);

  function logIn(email, password) {
    return signInWithEmailAndPassword(auth, email, password);
  }
  function signUp(email, password) {
    return createUserWithEmailAndPassword(auth, email, password);
  }
  function logOut() {
    return signOut(auth);
  }

  useEffect(() => {
    const unsubscribe = onAuthStateChanged(auth, (currentuser) => {
      if (currentuser) {
        setUser(currentuser.uid);
        const localStoredFamilyID = localStorage.getItem("familyID");
        const localStoredUserID = localStorage.getItem("userID");
        const localStoredRole = localStorage.getItem("userRole");
        setFamilyID(localStoredFamilyID);
        setUserID(localStoredUserID);
        setUserRole(localStoredRole);
      } else {
        setUser(null);
        setFamilyID(null);
        setUserID(null);
        setUserRole(null);
      }
    });
    return () => {
      unsubscribe();
    };
  }, []);
  return (
    <userAuthContext.Provider
      // eslint-disable-next-line react/jsx-no-constructed-context-values
      value={{
        user,
        logIn,
        signUp,
        logOut,
        chores,
        setChores,
        familyID,
        userID,
        userRole,
      }}
    >
      {children}
    </userAuthContext.Provider>
  );
};

export function useUserAuth() {
  return useContext(userAuthContext);
}
