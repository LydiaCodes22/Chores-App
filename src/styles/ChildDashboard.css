.child-dashboard-container {
    display: flex;
    flex-direction: column;
    align-items: center;
    align-self: center;
    background: var(--gradient);
    box-shadow: var(--shadow);
    padding: 3rem 1rem;
    border-radius: 20px;
    width: 90vw;
    max-width: 1000px;
    margin: 5rem;
  
    
    
  }
  .child-dashboard-container input {
    width: 20%;
    margin-right: 1rem;
    margin-left: 1rem;
    background-color: rgba(255, 255, 255, 0.466);
  }

  .child-dashboard-section {
    display: flex;
    flex-direction: column;
    gap: 15px;
    align-items: center;
    border-radius: 20px;
    border-color: var(--purple);
    border-style:solid;
    width: 80vw;
    max-width: 800px;
    padding: 1rem;
    margin-bottom: 2rem;
  }