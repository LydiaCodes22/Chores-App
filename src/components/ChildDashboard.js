/* eslint-disable react/prop-types */
import React, { useState, useEffect } from "react";
import axios from "axios";
import ChoreCard from "./ChoreCard";
import { useUserAuth } from "../context/UserAuthContext";
import "../styles/ChildDashboard.css";

const ChildDashboard = () => {
  const { chores, setChores, userID, familyID } = useUserAuth();
  useEffect(() => {
    if (familyID) {
      axios
        .get(`http://localhost:3300/family/${familyID}/chores`)
        .then((response) => {
          setChores(response.data);
        })
        .catch((e) => {
          console.log(e);
        });
    }
  }, [familyID]);
  const [requestedAmount, setRequestedAmount] = useState();
  const handleRequest = () => {
    // handle request function should POST the requested amount to the DB and reset the input field so it is blank again.
  };
  const handleFieldChange = (event) => {
    event.preventDefault();
    setRequestedAmount(event.target.value);
    console.log(requestedAmount);
  };
  return (
    <div className="container child-dashboard-container">
      <h1>Dashboard</h1>
      <div className="child-dashboard-section">
        <h2>
          Balance £0
          {
            // FUTURE DEVELOPMENT: to get this from the DB.
          }
        </h2>

        <form onSubmit={handleRequest}>
          <label htmlFor="requestedamount">
            Cash in your hard work - £
            <input
              type="number"
              name="requestedamount"
              onChange={handleFieldChange}
            />
            <button type="submit" className="btn btn-narrow btn-fill_peach">
              Request Amount
            </button>
          </label>
        </form>
      </div>
      <div className="child-dashboard-section">
        <h2>My Chores</h2>
        <button type="button" className="btn btn-narrow btn-stroke_purple">
          <a href="/findchore">+ Find a new chore!</a>
        </button>
        {chores
          // eslint-disable-next-line eqeqeq
          .filter((chore) => chore.owner == userID && chore.status == "T")
          .map((chore) => (
            <ChoreCard
              key={chore.choreID}
              name={chore.name}
              price={chore.price}
              choreID={chore.choreID}
              choreOwner={chore.owner}
              component="ChildDashboard"
            />
          ))}
      </div>
    </div>
  );
};

export default ChildDashboard;
