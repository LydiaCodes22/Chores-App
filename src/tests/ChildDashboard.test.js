/* eslint-disable react/jsx-no-constructed-context-values */
import { render, screen } from "@testing-library/react";
import React from "react";
import ChildDashboard from "../components/ChildDashboard";
import { userAuthContext } from "../context/UserAuthContext";

describe("the parents dashboard", () => {
  const validChore = {
    choreID: "1",
    name: "valid chore",
    price: "5",
    owner: "1",
  };
  const familyID = "";
  const chores = [validChore];
  const setChores = () => {};
  const userID = "1";
  // want to be able to test the chore is displayed

  it("matches the snapshot", () => {
    const { asFragment } = render(
      <userAuthContext.Provider value={{ familyID, chores, setChores, userID }}>
        <ChildDashboard />
      </userAuthContext.Provider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
  it("should render basic fields", () => {
    render(
      <userAuthContext.Provider value={{ familyID, chores, setChores, userID }}>
        <ChildDashboard />
      </userAuthContext.Provider>
    );
    expect(screen.getByText(/dashboard/i)).toBeInTheDocument();
    expect(
      screen.getByRole("button", { name: /find a new chore/i })
    ).toBeInTheDocument();
    expect(screen.getByText(/my chores/i)).toBeInTheDocument();
    expect(
      screen.getByRole("button", { name: /request amount/i })
    ).toBeInTheDocument();
  });
});
