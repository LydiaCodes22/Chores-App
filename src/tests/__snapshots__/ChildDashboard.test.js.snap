// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the parents dashboard matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="container child-dashboard-container"
  >
    <h1>
      Dashboard
    </h1>
    <div
      class="child-dashboard-section"
    >
      <h2>
        Balance £0
      </h2>
      <form>
        <label
          for="requestedamount"
        >
          Cash in your hard work - £
          <input
            name="requestedamount"
            type="number"
          />
          <button
            class="btn btn-narrow btn-fill_peach"
            type="submit"
          >
            Request Amount
          </button>
        </label>
      </form>
    </div>
    <div
      class="child-dashboard-section"
    >
      <h2>
        My Chores
      </h2>
      <button
        class="btn btn-narrow btn-stroke_purple"
        type="button"
      >
        <a
          href="/findchore"
        >
          + Find a new chore!
        </a>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
